pulsar-metadata/src/main/java/org/apache/pulsar/metadata/coordination/impl/ResourceLockImpl.java

```diff
@@ -196,6 +196,7 @@ synchronized void lockWasInvalidated() {
                             // We failed to revalidate the lock due to connectivity issue
```

Comment:
Plus, I think we also need to handle `LockBusyException` here to avoid forgetting to complete `expiredFuture`.